bundles/org.openhab.binding.proteusecometer/src/main/java/org/openhab/binding/proteusecometer/internal/ecometers/handler/ProteusEcoMeterSHandler.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.proteusecometer.internal.ecometers.handler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.proteusecometer.internal.ProteusEcoMeterBindingConstants;
+import org.openhab.binding.proteusecometer.internal.ProteusEcoMeterConfiguration;
+import org.openhab.binding.proteusecometer.internal.WrappedException;
+import org.openhab.binding.proteusecometer.internal.ecometers.ProteusEcoMeterSReply;
+import org.openhab.binding.proteusecometer.internal.ecometers.ProteusEcoMeterSService;
+import org.openhab.binding.proteusecometer.internal.serialport.SerialPortService;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fazecast.jSerialComm.SerialPort;
+
+/**
+ * The {@link ProteusEcoMeterSHandler} updates thing channels when receiving data
+ *
+ * @author Matthias Herrmann - Initial contribution
+ */
+@NonNullByDefault
+public class ProteusEcoMeterSHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ProteusEcoMeterSHandler.class);
+    private @Nullable SerialPort serialPort;
+    private @Nullable ProteusEcoMeterConfiguration config;
```

Comment:
```suggestion
    private ProteusEcoMeterConfiguration config = new ProteusEcoMeterConfiguration ();
```
Remove the need to check for null all the time.